src/util/simplify_expr2.cpp

```diff
@@ -1912,7 +1912,27 @@ static expr2tc simplify_relations(
       return to_constant_floatbv2t(c).value;
     };
 
-    simpl_res = TFunctor<ieee_floatt>::simplify(
+    simpl_res = TFunctor<ieee_floatt &>::simplify(
+      simplied_side_1, simplied_side_2, is_constant, get_value);
+  }
+  else if(is_pointer_type(simplied_side_1) || is_pointer_type(simplied_side_2))
+  {
+    std::function<bool(const expr2tc &)> is_constant =
+      [&](const expr2tc &t) -> bool {
+      if(is_pointer_type(t) && is_symbol2t(t))
+      {
+        symbol2t s = to_symbol2t(t);
+        if(s.thename == "NULL")
+          return true;
+      }
+      return false;
+    };
+
+    std::function<int(expr2tc &)> get_value = [](expr2tc &) -> int {
+      return 0xbadbeef;
+    };
+
+    simpl_res = TFunctor<int>::simplify(
```

Comment:
Shoudn't this be moved to another PR?

Reply:
I see, could you update your comment to reflect this then? Also, could you add testcases with `--no-simplify`?

Reply:
sure!